cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperatorTest.java

```diff
@@ -90,65 +99,661 @@
                     .build();
 
     @BeforeAll
-    public static void before() throws IOException, URISyntaxException {
+    public static void beforeAll() throws IOException, URISyntaxException {
         ccServer = MockCruiseControl.getCCServer(CruiseControl.REST_API_PORT);
     }
 
     @AfterAll
-    public static void after() {
+    public static void afterAll() {
         ccServer.stop();
     }
 
     @BeforeEach
-    public void resetServer() {
+    public void beforeEach(Vertx vertx) {
         ccServer.reset();
+
+        kubernetesClient = new MockKube()
+                .withCustomResourceDefinition(Crds.kafkaRebalance(), KafkaRebalance.class, KafkaRebalanceList.class, DoneableKafkaRebalance.class)
+                .end()
+                .build();
+
+        ResourceOperatorSupplier supplier = ResourceUtils.supplierWithMocks(true);
+        PlatformFeaturesAvailability pfa = new PlatformFeaturesAvailability(true, kubernetesVersion);
+        kcrao = new KafkaRebalanceAssemblyOperator(vertx, pfa, supplier, HOST);
+
+        mockRebalanceOps = supplier.kafkaRebalanceOperator;
+        mockKafkaOps = supplier.kafkaOperator;
+    }
+
+    @AfterEach
+    public void afterEach() {
+        if (kubernetesClient != null) {
+            kubernetesClient.close();
+        }
     }
 
     @Test
-    public void testNewRebalance(Vertx vertx, VertxTestContext context) throws IOException, URISyntaxException {
+    public void testNewToProposalReadyRebalance(VertxTestContext context) throws IOException, URISyntaxException {
```

Comment:
I think a javadoc description of precisely the scenario being tested for each of the tests in the class might prove useful when maintaining them in the future.